### PR TITLE
feat(generator): remove allowed reward proxies

### PR DESCRIPTION
**Pull request by P-Yevhenii**

### CODE REVIEW

**Review by ShadoySV — APPROVED** [review]

File: contracts/tokenomics/generator/src/migration.rs [hunk 1]
```diff
@@ -129,6 +129,39 @@ pub struct ConfigV200 {
...
+    /// Contract address which can only set active generators and their alloc points
+    pub generator_controller: Option<Addr>,
+    /// The voting escrow contract address
+    pub voting_escrow: Option<Addr>,
+    /// The voting escrow delegation contract address
+    pub voting_escrow_delegation: Option<Addr>,
+    /// The ASTRO token address
+    pub astro_token: Addr,
+    /// Total amount of ASTRO rewards per block
+    pub tokens_per_block: Uint128,
```
Comment (epanchee) [inline contracts/tokenomics/generator/src/migration.rs:148]: _allowed_reward_proxies_ field is missed

**Review by epanchee — CHANGES_REQUESTED** [review]

**Review by epanchee — APPROVED** [review]

**Review by ShadoySV — CHANGES_REQUESTED** [review]
We need an endpoint for querying all unique proxies ever used

File: contracts/tokenomics/generator/src/contract.rs [hunk 2]
```diff
@@ -1450,10 +1450,21 @@ fn init_proxy_rewards_holder(
     Ok(msg)
 }
 
+/// Adds a new proxy contract to the list
+fn update_reward_proxies(deps: DepsMut, proxy: &Addr) -> Result<(), ContractError> {
+    let mut all_proxies = REWARD_PROXIES_LIST.load(deps.storage)?;
+    if !all_proxies.contains(proxy) {
+        all_proxies.push(proxy.clone());
+        REWARD_PROXIES_LIST.save(deps.storage, &all_proxies)?;
```
Comment (epanchee) [inline contracts/tokenomics/generator/src/contract.rs:1458]: We have the proxies list [already](https://github.com/astroport-fi/astroport-core/blob/1b94b050c6d1ca8315f14df7a16e3b963fd8f388/contracts/tokenomics/generator/src/state.rs#L152). I suggest to use `PROXY_REWARD_ASSET.keys()` and remove REWARD_PROXIES_LIST and this function.

**Review by epanchee — CHANGES_REQUESTED** [review]

File: contracts/tokenomics/generator/src/contract.rs (hunk 2 above)
Comment (ShadoySV) [reply contracts/tokenomics/generator/src/contract.rs:1458]: Yes, but ```keys``` method will deserialize values [too](https://github.com/CosmWasm/cw-plus/blob/580e555dd091fd7d80c0e291f4c65fd418e87981/packages/storage-plus/src/prefix.rs#L179), thus limiting maximum possible elements in the array due to query gas limitation. Something similar is with storing this array into the config

Comment (ShadoySV) [reply contracts/tokenomics/generator/src/contract.rs:1458]: Ok, will do for code clarity and gas saving at the beginning

**Review by epanchee — APPROVED** [review]

**Review by ShadoySV — APPROVED** [review]

**Pull request merged**